app/src/main/java/com/arctouch/codechallenge/data/MovieBasePagedKeyedDataSource.java

```diff
@@ -0,0 +1,18 @@
+package com.arctouch.codechallenge.data;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.paging.PageKeyedDataSource;
+
+import com.arctouch.codechallenge.data.model.Movie;
+
+public abstract class MovieBasePagedKeyedDataSource extends PageKeyedDataSource<Integer, Movie> {
+    private MutableLiveData<Throwable> error;
+
+    public MovieBasePagedKeyedDataSource() {
+        error = new MutableLiveData<>();
+    }
+
+    public MutableLiveData<Throwable> getError() {
```

Comment:
In my opinion, it is better to have an object variable as private because you maintain better encapsulation in your code. The way you suggested, even if the attribute is final I would be letting another person access this variable directly and as I sad it hurts Encapsulation.